src/test/java/it/pagopa/ecommerce/payment/methods/service/PaymentMethodServiceTests.java

```diff
@@ -341,7 +344,21 @@ void shouldRetrieveCardDataForInvalidPaymentMethodId() {
     }
 
     @Test
-    void shouldRetrieveCardDataWithSuccess() {
+    void shouldRetrieveCardDataForInvalidSessionId() {
+        String paymentMethodId = "paymentMethodId";
+        String sessionId = "sessionId";
+        PaymentMethod paymentMethod = TestUtil.getPaymentMethod();
+        PaymentMethodDocument paymentMethodDocument = TestUtil.getTestPaymentDoc(paymentMethod);
+        Mockito.when(paymentMethodRepository.findById(paymentMethodId)).thenReturn(Mono.just(paymentMethodDocument));
+        Mockito.when(npgSessionsTemplateWrapper.findById(sessionId)).thenReturn(Optional.empty());
+        StepVerifier.create(paymentMethodService.getCardDataInformation(paymentMethodId, any()))
+                .expectErrorMatches(e -> e instanceof SessionIdNotFoundException)
+                .verify();
+
```

Comment:
Add mockito verify assertions against redis wrapper and npgClient mock method calls
```suggestion
        Mockito.verify(npgSessionsTemplateWrapper, Mockito.times(1)).findById(any());
        Mockito.verify(npgSessionsTemplateWrapper, Mockito.times(0)).save(any());
        Mockito.verify(npgClient, Mockito.times(0)).getCardData(any(), any());
```

Reply:
Fixed with ccf8910831947ede3edd4a71e5002d8aa12e7f1b